### PR TITLE
20230415 web config addons

### PR DESCRIPTION
**Pull request by TheTrainGoes**

### CODE REVIEW

File: docs/web-configurator.md [hunk 1]
```diff
@@ -58,26 +58,69 @@ If you have a setup with per-button RGB LEDs, they can be configured here.
...
+
+![GP2040 Configurator - Add-Ons BOOTSEL Button Configuration](assets/images/gpc-add-ons-bootsel-button.png)
+
+* `BOOTSEL Button` - Choose an input to be mapped tot he BOOTSEL button. 
+
+Please note that this can only be used on devices that have a BOOTSEL button.  Please also note that the OLED might become unresponsive if this button is set.  You can unset it to restore OLED fuctionality.
+
+On-board LED Configuration
+
+![GP2040 Configurator - Add-Ons On-Board LED Configuration](assets/images/gpc-add-ons-onboard-led.png)
```
Comment (FeralAI) [inline docs/web-configurator.md:86]: Link is broken due to missing image.

File: docs/web-configurator.md [hunk 2]
```diff
@@ -58,26 +58,69 @@ If you have a setup with per-button RGB LEDs, they can be configured here.
...
-* `Splash Choice` - Choose the splash screen that is displayed when the unit is turned on.
+* `Splash Duration` - Sets the amount of time the splash screen displays for on boot.
+* `Display Saver Timeout` - Will cause the display to turn off afer the specified number of minuites.  Pressing any input will cause the diplay to turn back on.
+* `Choose File` - This will allow you to upload your own image to be used for the splash screen.  It is recommend that you use a two color 128x64 image (or one that is sized appropiatly for your display).  Uploading any other type of image will result in a conversion and sizing of the image automatically.  If the image is inverted upon upload, just check off the `Invert` box.
 
 Check out our collection of great custom splash screens from the community [HERE](community-splash-screens.md)
 
 ## Add-Ons Configuration
 
 This section is for custom add-ons that can be enabled to exand the functionality of GP2040-CE.  Please not that not all add-ons may work together.  These should be considered experminental.
```
Comment (FeralAI) [inline docs/web-configurator.md:74]: `exand` should be `expand`. `Please not that` should be `Please note that`. `experminental` should be `experimental`.

File: docs/web-configurator.md [hunk 3]
```diff
@@ -58,26 +58,69 @@ If you have a setup with per-button RGB LEDs, they can be configured here.
...
 
+BOOTSEL Button Configuration
```
Comment (FeralAI) [inline docs/web-configurator.md:76]: The addon headers don't stand out on the website. Maybe these should be H3's? with `### BOOTSEL Button Configuration`?

File: docs/web-configurator.md [hunk 4]
```diff
@@ -58,26 +58,69 @@ If you have a setup with per-button RGB LEDs, they can be configured here.
...
+* `Turbo Always On 4` - The GPIO pin used for a Turbo button that will always be on.
+* `Charge Button 1 Pin` - The GPIO pin used for a button that needs to be able to do a charged shot.
+* `Charge Button 2 Pin` - The GPIO pin used for a button that needs to be able to do a charged shot.
+* `Charge Button 3 Pin` - The GPIO pin used for a button that needs to be able to do a charged shot.
+* `Charge Button 4 Pin` - The GPIO pin used for a button that needs to be able to do a charged shot.
+* `Charge Button 1 Assignment` - The button that will be able to charge shot regardless of Turbo status.
+* `Charge Button 2 Assignment` - The button that will be able to charge shot regardless of Turbo status.
+* `Charge Button 3 Assignment` - The button that will be able to charge shot regardless of Turbo status.
+* `Charge Button 4 Assignment` - The button that will be able to charge shot regardless of Turbo status.
+* `SimSimultaneous Priority Mode` - In the event both the Turbo and charged buttons are pressed at the time same, which should take priority.
```
Comment (FeralAI) [inline docs/web-configurator.md:123]: Typo in `SimSimultaneous`

File: docs/web-configurator.md [hunk 5]
```diff
@@ -107,6 +150,17 @@ I2C Analog ADS1219
...
+Dual Directional Input
+
+![GP2040 Configuration - Add-Ons Dual Directional Input](assets/images/gpc-add-ons-dual-directional.png)
+
+* `Dual Up Pin` - The GPIO pin used for the secondary Up direction.
+* `Dual Down Pin` - The GPIO pin used for the secondary Down direction.
+* `Dual Left Pin` - The GPIO pin used for the secondary Left direction.
+* `Dual Right Pin` - The GPIO pin used for the secondary Right direction.
+* `Dual D-Pad Mode` - Choose if this should act as an additional instance of the D-Pad or as the Left or Right stick.
+* `Combination Mode` - Choose how these inputs should be combined.
```
Comment (FeralAI) [inline docs/web-configurator.md:162]: Can we define the options here?

File: docs/web-configurator.md [hunk 6]
```diff
@@ -115,12 +169,32 @@ Buzzer Speaker
...
+* `Extra Button Pin` - The GPIO pin used for the extra instance of a button.
+* `Extra Button` - The button that will have an extra instance.
+
+Player Number (X-INPUT ONLY)
+
+![GP2040 Configurator - Player Number](assets/images/gpc-add-ons-player-number.png)
+
+* `Player Number` - Choose what player number this RP2040 based device will be representing.  This is only for X-INPUT mode.
+
+SOCD Selection Slider
```
Comment (FeralAI) [inline docs/web-configurator.md:185]: Looks like some of the SOCD content is in the Data Backup and Restoration section.

**Review by FeralAI — COMMENTED** [review]
Found some issues, a few were already there but I flagged them anyway.

File: docs/web-configurator.md (hunk 1 above)
Comment (TheTrainGoes) [reply docs/web-configurator.md:86]: Updated thank you

File: docs/web-configurator.md (hunk 2 above)
Comment (TheTrainGoes) [reply docs/web-configurator.md:74]: Updated thank you.

File: docs/web-configurator.md (hunk 3 above)
Comment (TheTrainGoes) [reply docs/web-configurator.md:76]: Will chat with you more about this one.

File: docs/web-configurator.md (hunk 4 above)
Comment (TheTrainGoes) [reply docs/web-configurator.md:123]: Updated thank you

**Review by FeralAI — APPROVED** [review]

**Review by FeralAI — APPROVED** [review]
Looks good

**Review by arntsonl — APPROVED** [review]
Looks great!!! Thanks Feral for reviewing this as well!

**Pull request merged**